src/app/phase1/issue/issue.component.ts

```diff
@@ -50,10 +52,18 @@ export class IssueComponent implements OnInit {
   }
 
   private getIssue(id: number) {
-    this.issueService.getIssue(id).pipe(finalize(() => this.isIssueLoading = false)).subscribe((issue) => {
-      this.issue = issue;
+    this.issueSubscription = this.issueService.getIssues().subscribe((issues) => {
+      if (issues !== undefined) {
```

Comment:
I dont think issues will ever be undefined over here. So there is no need for this check. If you look at `IssueService.getIssues()`, if `this.issues` is undefined, if will make an API call to fetch all the issues. Same goes for other phases that was done in the same way.

Reply:
When I tried to Log out from the edit issue page, it throws an error that tells me that `this.issues` is undefined because I have reset it.

Reply:
Yup I have removed this condition